pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarClientImpl.java

```diff
@@ -83,6 +83,7 @@
     private final ConnectionPool cnxPool;
     private final Timer timer;
     private final ExecutorProvider externalExecutorProvider;
+    private Map<String, Schema> supportSchemaVersioningSchemaCache = new HashMap<>();
```

Comment:
I think the topic is the first key, the true avro schema is second key. like this 

private Map<String, ConcurrentHashMap<org.apache.avro.Schema,Schema>> supportSchemaVersioningSchemaCache = new HashMap<>();
Because schema by version will create a lot of same reader and writer